x-pack/plugins/security_solution/public/management/pages/trusted_apps/store/selectors.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Immutable } from '../../../../../common/endpoint/types';
+import { TrustedAppsPageState } from '../state/trusted_apps_page_state';
+import { pageInfosEqual } from '../state/items_page';
+import { isStaleBinding } from '../state/async_data_binding';
+
+export const needsRefreshOfListData = (state: Immutable<TrustedAppsPageState>) => {
```

Comment:
I would like to suggest to revisit this in case there are performance problems, I started doing this but it makes the code more complex to read to be honest. At least I would suggest to do it as a part of some next stories. Please reopen this conversation if you disagree.